src/GitHub.Exports/Models/DiffUtilities.cs

```diff
@@ -116,61 +118,72 @@ public static DiffLine Match(IEnumerable<DiffChunk> diff, IList<DiffLine> target
 
         public class LineReader
         {
+            static readonly LineInformation Default = new LineInformation(null, 0);
+            static readonly LineInformation Empty = new LineInformation(String.Empty, 0);
             readonly string text;
+            readonly int length = 0;
             int index = 0;
 
             public LineReader(string text)
             {
+                Guard.ArgumentNotNull(text, nameof(text));
                 this.text = text;
+                this.length = text.Length;
             }
 
-            public string ReadLine()
+            public LineInformation ReadLine()
             {
                 if (EndOfText)
                 {
                     if (StartOfText)
                     {
                         index = -1;
-                        return string.Empty;
+                        return Empty;
                     }
-
-                    return null;
+                    return Default;
                 }
 
-                var startIndex = index;
-                index = text.IndexOf('\n', index);
-                var endIndex = index != -1 ? index : text.Length;
-                var length = endIndex - startIndex;
-
-                if (index != -1)
+                var carriageReturns = 0;
+                StringBuilder sb = new StringBuilder();
+                for (; index < length; index++)
```

Comment:
We could maybe adapt this implementation to use, `text.IndexOfAny(new[] {'\r', '\n'}, index)`? With `new[] {'\r', '\n'}`, stored in a static. 😉